### PR TITLE
Fix/Storage Group expiration

### LINKED ISSUE #2016: Storage group wasn't deleted after lifetime was expired
## Expected Behavior
Storage group should be deleted after lifetime expired

## Current Behavior
Storage group wasn't deleted after lifetime was expired

## Steps to Reproduce (for bugs)
1. Create storagegroup with lifetime=1
```
COMMAND: neofs-cli --config /jenkins/workspace/sbercloud_test/tmp.nzkHtw9gNy/neofs-testcases/wallet_config.yml storagegroup put --rpc-endpoint '172.26.160.37:8080' --wallet '/jenkins/workspace/sbercloud_test/tmp.nzkHtw9gNy/neofs-testcases/TemporaryDir/9a64eee7-52c1-4ee2-848e-11fc954f6550.json' --cid '7QwoRQ8cW63GRyT5DZJbXcpkBhyaCHsPsUBvV6PGQAHt' --members '3nDYw6tVNE15HsqgJHcfk8uLHHLfeLoysp6kZmysUxii' --lifetime 1
RETCODE: 0

STDOUT:
Storage group successfully stored
  ID: HRCKtbDHaVXcb51F3vde89fFw99E9VfZXoWd8zdTeGCu
  CID: 7QwoRQ8cW63GRyT5DZJbXcpkBhyaCHsPsUBvV6PGQAHt

STDERR:

Start / End / Elapsed	 08:10:27.037715 / 08:10:27.474764 / 0:00:00.437049
```
2. Tick epoch 2 times
```
COMMAND: neofs-adm --config /jenkins/workspace/sbercloud_test/tmp.nzkHtw9gNy/tatlin-object-testsetup/.setup/config.yaml morph force-new-epoch 
RETCODE: 0

STDOUT:
Current epoch: 100, increase to 101.
Waiting for transactions to persist...

STDERR:
Start / End / Elapsed	 08:10:27.476317 / 08:10:39.208320 / 0:00:11.732003

COMMAND: neofs-adm --config /jenkins/workspace/sbercloud_test/tmp.nzkHtw9gNy/tatlin-object-testsetup/.setup/config.yaml morph force-new-epoch 
RETCODE: 0

STDOUT:
Current epoch: 101, increase to 102.
Waiting for transactions to persist...

STDERR:
Start / End / Elapsed	 08:10:39.209956 / 08:10:46.896181 / 0:00:07.686225
```
3. Try to get storage group
```
COMMAND: neofs-cli --config /jenkins/workspace/sbercloud_test/tmp.nzkHtw9gNy/neofs-testcases/wallet_config.yml storagegroup get --rpc-endpoint '172.26.160.37:8080' --wallet '/jenkins/workspace/sbercloud_test/tmp.nzkHtw9gNy/neofs-testcases/TemporaryDir/9a64eee7-52c1-4ee2-848e-11fc954f6550.json' --cid '7QwoRQ8cW63GRyT5DZJbXcpkBhyaCHsPsUBvV6PGQAHt' --id 'HRCKtbDHaVXcb51F3vde89fFw99E9VfZXoWd8zdTeGCu'
RETCODE: 0

STDOUT:
Expiration epoch: 101
Group size: 1000
Group hash: <empty>
Members:
	3nDYw6tVNE15HsqgJHcfk8uLHHLfeLoysp6kZmysUxii

STDERR:
Start / End / Elapsed	 08:10:46.897948 / 08:10:47.281773 / 0:00:00.383825
```
**Logs:**[d0bed81bfd1c6fc4.zip](https://github.com/nspcc-dev/neofs-node/files/9927724/d0bed81bfd1c6fc4.zip)
**Versions:**
```
NeoFS node 0.33.0-78-g846ff515
```
## Your Environment
Server setup and configuration:
cloud, 4 VMs, 4 SN, 4 http qw, 4 s3 gw

Operating System and version (uname -a):
linux vedi 5.10.0-16-amd64 https://github.com/nspcc-dev/neofs-node/issues/1 SMP Debian 5.10.127-1 (2022-06-30) x86_64 GNU/Linux

### PR DESCRIPTION
**Pull request by carpawell**
Closes #2016.

### CODE REVIEW

File: pkg/local_object_storage/writecache/writecache.go [hunk 1]
```diff
@@ -170,3 +170,7 @@ func (c *cache) Close() error {
 	}
 	return nil
 }
+
+func (c *cache) typ() string {
+	return "write-cache"
+}
```
Comment (fyrchik) [inline pkg/local_object_storage/writecache/writecache.go:176]: Please, can we make it a constant?

File: pkg/local_object_storage/shard/head.go [hunk 2]
```diff
@@ -46,26 +46,25 @@ func (r HeadRes) Object() *objectSDK.Object {
...
-		}
-
-		// otherwise object seems to be flushed to metabase
-	}
-
 	var obj *objectSDK.Object
 	var err error
+
 	if s.GetMode().NoMetabase() {
+		if s.hasWriteCache() {
```
Comment (fyrchik) [inline pkg/local_object_storage/shard/head.go:53]: It make no sense, in degraded mode writecache is unavailable too. I think we can just drop this.

File: pkg/local_object_storage/writecache/writecache.go (hunk 1 above)
Comment (fyrchik) [reply pkg/local_object_storage/writecache/writecache.go:176]: I mean we don't need a method for this.

Comment (carpawell) [reply pkg/local_object_storage/writecache/writecache.go]: fixed

File: pkg/local_object_storage/shard/head.go (hunk 2 above)
Comment (carpawell) [reply pkg/local_object_storage/shard/head.go]: dropped

> in degraded mode writecache is unavailable too

made me recheck that fact. isnt obvious, mb that a problem

**Review by acid-ant — APPROVED** [review]

File: pkg/local_object_storage/shard/head.go [hunk 3]
```diff
@@ -46,25 +44,9 @@ func (r HeadRes) Object() *objectSDK.Object {
...
-			// in this case we think that object is presented in write-cache, but corrupted
-			return HeadRes{}, fmt.Errorf("could not read header from write-cache: %w", err)
-		}
-
-		// otherwise object seems to be flushed to metabase
-	}
-
 	var obj *objectSDK.Object
 	var err error
+
```
Comment (fyrchik) [inline pkg/local_object_storage/shard/head.go:49]: empty line

**Review by fyrchik — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]

**Pull request merged**